src/protocol.rs

```diff
@@ -101,13 +107,24 @@ impl Participant {
     /// Pulls the first message from the participant's inbox, and then
     /// potentially outputs a bunch of messages that need to be delivered to
     /// other participants' inboxes.
+    ///
+    /// ## Return type
+    /// This returns a tuple of a session ID, an output, and a set of messages.
+    /// - The session ID will always match the session ID of the input message,
+    ///   and of all the outgoing messages. It's passed out as a convenience for
+    ///   associating with the [`Output`].
+    /// - The [`Output`] is the output of the protocol with the given session
+    ///   ID, if this message happened to terminate the protocol for this
```

Comment:
Minor: I found the "if this message..." portion confusing, because I would have expected `Option<Output>` as the type but rather non-termination is encoded using `Output::None`, so `Output` encodes the output of the protocol _regardless_ of whether the protocol terminates, right? If so, maybe remove this part of the documentation?

Reply:
Oh, good call. I rephrased.

FWIW I debated about doing an option vs this approach. We might change it later depending on how the example flows.